react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { password: '' };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ password: event.target.value });
+    const $err = document.querySelector('.errorMsg');
+    const $icon = document.querySelector('.icon');
+    if (event.target.value.length < 8) {
+      $err.innerHTML = 'Password must be 8 characters.';
+      $icon.innerHTML = '❌';
+    } else if (event.target.value.length === 0) {
+      $err.innerHTML = 'A password is required.';
+      $icon.innerHTML = '❌';
+    } else {
+      $err.innerHTML = '';
+      $icon.innerHTML = '✅';
+    }
```

Comment:
We never want to do direct DOM manipulation with React. A better approach would be to have these checks in the `render` method, set the values to a variable, then use those variables in your JSX to determine what is rendered.

Reply:
awesome thank you, i knew there was something...

looks much better now